src/StoreApi/Schemas/CartSchema.php

```diff
@@ -404,14 +404,28 @@ protected function get_tax_lines( $cart ) {
 	 */
 	protected function get_cart_errors( $cart ) {
 		$controller    = new CartController();
-		$item_errors   = array_filter(
-			$controller->get_cart_item_errors(),
+		$cart_errors   = [];
+		$item_errors   = $controller->get_cart_item_errors();
+		$coupon_errors = $controller->get_cart_coupon_errors();
+
+		try {
+			/**
+			 * Fire action to validate cart. Functions hooking into this should throw a \RouteException.
+			 *
+			 * @param \WC_Cart $cart Cart object.
+			 */
+			do_action( 'wooocommerce_store_api_validate_cart', $cart );
```

Comment:
I see checkout route calls  `woocommerce_check_cart_items` which can also be used to add notices and have them converted to exceptions. Is that more suitable?

I also see we already have a `wooocommerce_store_api_validate_cart_item` action which puts error messages into this array. Why is a new hook needed which does the same thing?